qa/tasks/cbt.py

```diff
@@ -58,23 +78,71 @@ def install_dependencies(self):
-        self.first_mon.run(
-            args=[
-                'git', 'clone', '-b', 'master',
-                'https://github.com/axboe/fio.git',
-                '{tdir}/fio'.format(tdir=testdir)
-            ]
-        )
-        self.first_mon.run(
-            args=[
-                'cd', os.path.join(testdir, 'fio'), run.Raw('&&'),
-                './configure', run.Raw('&&'),
-                'make'
-            ]
-        )
+
+        benchmark_type = self.cbt_config.get('benchmarks').keys()[0]
+        self.log.info('benchmark: %s', benchmark_type)
+
+        if benchmark_type == 'librbdfio':
+            # install fio
+            testdir = misc.get_testdir(self.ctx)
+            self.first_mon.run(
+                args=[
+                    'git', 'clone', '-b', 'master',
+                    'https://github.com/axboe/fio.git',
+                    '{tdir}/fio'.format(tdir=testdir)
+                ]
+            )
+            self.first_mon.run(
+                args=[
+                    'cd', os.path.join(testdir, 'fio'), run.Raw('&&'),
+                    './configure', run.Raw('&&'),
+                    'make'
+                ]
+            )
+
+        if benchmark_type == 'cosbench':
+            # install cosbench
+            self.log.info('install dependecies for cosbench')
+            if system_type == 'rpm':
+                cosbench_depends = ['wget', 'unzip', 'java-1.7.0-openjdk', 'curl']
+            else:
+                cosbench_depends = ['wget', 'unzip', 'openjdk-8-jre', 'curl']
+            self.first_mon.run(args=install_cmd + cosbench_depends)
+            testdir = misc.get_testdir(self.ctx)
+            scripts = ['cli.sh', 'cosbench-start.sh', 'cosbench-stop.sh', 'start-all.sh', 'start-controller.sh', 'start-driver.sh',  'stop-all.sh', 'stop-controller.sh', 'stop-driver.sh']
+            self.first_mon.run(
+                args=[
+                    'cd', testdir, run.Raw('&&'),
+                    'wget',
+                    'https://github.com/intel-cloud/cosbench/releases/download/v0.4.2.c3/0.4.2.c3.zip', run.Raw('&&'),
+                    'unzip', '0.4.2.c3.zip', run.Raw('&&'),
+                    'cd', '0.4.2.c3',
+                ]
+            )
+            for script in scripts:
+                self.first_mon.run(
+                args=[
+                    'cd', os.path.join(testdir, '0.4.2.c3'), run.Raw('&&'),
+                    'chmod', '+x', script,
+                ]
+            )
+            # start cosbench and check info
+            self.log.info('start cosbench')
+            self.first_mon.run(
+                args=[
+                    'cd', testdir, run.Raw('&&'),
+                    'cd', '0.4.2.c3', run.Raw('&&'),
+                    'sh', 'start-all.sh'
+                ]
+            )
+            self.log.info('check cosbench info')
+            self.first_mon.run(
+                args=[
+                    'cd', testdir, run.Raw('&&'),
```

Comment:
could store testdir/0.4.2.c3 in a local var so it's easier to change later for each of these calls if we change versions